src/main/scala/net/snowflake/spark/snowflake/SparkConnectorContext.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2015-2020 Snowflake Computing
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.snowflake.spark.snowflake
+
+import net.snowflake.spark.snowflake.io.SnowflakeResultSetRDD.WORKER_LOG_PREFIX
+
+object SparkConnectorContext {
+  // TODO: need to merge with another PR to use LoggerWrapper.
+  import org.slf4j.LoggerFactory
+  private[snowflake] val logger = LoggerFactory.getLogger(getClass)
+
+  private var isExecutorConfigLogged = false
+
+  // One executor may execute multiple tasks, it is enough to log once.
+  private[snowflake] def executorLogConfigIfNotYet(): Unit = {
```

Comment:
You should rename this function.
recordConfig.
Also, why not just use this same call to store the telemetry version of the config you want to send.

Reply:
I will rename it.
Could you please explain below comments?
"Also, why not just use this same call to store the telemetry version of the config you want to send."

Reply:
This log is for executor only. I rename it as recordConfigOnExecutor() finally.